File: PYTHON/PROGRAMMERS/131701.py
# 연속 부분 수열 합의 개수
# lv 2
# 구현

def solution(elements):
    arr = []
    l = len(elements)
    for i in range(l):
        for j in range(l - 1):
            arr.append(sum(elements[i:i + j + 1]))
        elements.append(elements[i])
    
    arr = set(arr)
    return len(arr) + 1

File: JAVASCRIPT/PROGRAMMERS/131701.js
// 연속 부분 수열의 합
// lv 2
// 구현
function solution(elements) {
  var answer = 0;
  let arr = [];
  let l = elements.length;

  for (let i = 0; i < l; i++) {
    for (let j = 0; j < l - 1; j++) {
      let sum = elements.slice(i, i + j + 1).reduce((acc, cur) => acc + cur);
      arr.push(sum);
    }
    elements.push(elements[i]);
  }
  let setArr = new Set(arr);
  answer = setArr.size;
  return answer + 1;
}
